dipy/reconst/shm.py

```diff
@@ -1060,8 +1060,12 @@ def anisotropic_power(sh_coeffs, norm_factor=0.00001, power=2,
         ap += ap_i
         n_start = n_stop
 
-    # Shift the map to be mostly non-negative:
-    log_ap = np.log(ap) - np.log(norm_factor)
+    # Shift the map to be mostly non-negative,
+    # only applying the log operation to non-zero elements
```

Comment:
Negative log(ap) is not really a problem, for 2 reasons:

1. The ```-log(norm_factor)``` will shift most of them back to positive values, as long as the norm_factor stays reasonable. For example, the default value is 0.00001, for which the ```-log(norm_factor)``` will result in ```+11.5129```, shifting all values by that value.

2. If the ```non_negative``` arg is set to True (which is the default), there is the additional check later in the function that can handle all remaining negavtive values.